gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala

```diff
@@ -164,7 +164,7 @@ trait BasicScanExecTransformer extends TransformSupport {
     // Will put all filter expressions into an AND expression
     val transformer = filterExprs()
       .reduceLeftOption(And)
-      .map(ExpressionConverter.replaceWithExpressionTransformer(_, output))
+      .map(ExpressionConverter.replaceWithExpressionTransformer(_, output, false))
```

Comment:
How do we decide this config to be true or not?

Reply:
`ExpressionConverter.replaceWithExpressionTransformer` will be invoked in two places of an operator.
One is when we do `doValidate`, and the other is `doTransform`.

If it's invoked in `doValidate`, the parameter is **true**, which means that we don't need to really trigger the execution of subquery, the `ScalarSubqueryTransformer.doTransform` just return a fake ExpressionNode

if it's invoked in `doTransform`, the parameter is **false**, which will trigger the execution of scala subquery and get the real ExpressionNode

Reply:
Thanks for your reply! From my understanding, this parameter is mainly used to ditinguish ScalarSubqueryTransformer's output. If we make this paramater as a construction parameter of ExpressionConverter, would this part of logic become simpler? In this way, we don't need to add `validation` to each `replaceWithExpressionTransformer` method.

Reply:
That's a good idea. Let me update it

Reply:
updated

Reply:
Thank you!